src/WebApp/index.tsx

```diff
@@ -94,15 +94,21 @@ async function handleResponse(
 ): Promise<MessageResponse> {
   const { action, result, ruid , error : msgerror} = data;
   if (msgerror) {
-    throw new Error(msgerror);
+    resolverMap.delete(ruid);
+    rejectMap.get(ruid)(msgerror);
+    rejectMap.delete(ruid);
+    return { ruid, action, result: 'done' };
```

Comment:
is it intended to return success here instead of throwing error? any reason?

Reply:
the response error is handled by rejecting the promise .
we actually do not need to return anything for handling responses
I will remove the return type

throw error (at the end) only when the action is not `handled`